test/widgets/interactive_board_test.dart

```diff
@@ -0,0 +1,150 @@
+import 'package:chessground/chessground.dart';
+import 'package:dartchess/dartchess.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:lichess_mobile/src/model/settings/board_preferences.dart';
+import 'package:lichess_mobile/src/widgets/interactive_board.dart';
+import '../test_helpers.dart';
+import '../test_provider_scope.dart';
+
+void main() {
+  final setup = Setup.parseFen('r3k2r/8/8/8/8/8/8/R3K2R w KQkq - 0 1');
+  Position pos = Chess.fromSetup(setup);
+
+  final gameData = GameData(
+    playerSide: PlayerSide.white,
+    sideToMove: Side.white,
+    validMoves: makeLegalMoves(pos),
+    promotionMove: null,
+    onMove: (NormalMove move, {bool? isDrop}) {
+      //This doesn't work because the position is not updated in the UI
```

Comment:
For this to work it would need to be part of a StatefulWidget and be called inside `setState`.

Ideally we'd want to have an higher level test on a whole screen, and that would solve this. This is preferable because if we refactor and remove the `InteractiveBoardWidget`, the tests for the whole screen will still work and be relevant.

You could for instance test this on the `AnalysisScreen`, because it is easy to build a test screen with a PGN that gives you a position ready to castle.

You could add a
```dart
group('respect castling preferences', () {
   // add your tests here
});
```

I can take care of the GameScreen tests if you want, since it's a little bit harder to setup (need to mock websocket response). There are already some examples on how to do this though.